pkg/skaffold/initializer/init.go

```diff
@@ -290,6 +290,9 @@ func generateSkaffoldPipeline(k Initializer, dockerfilePairs []dockerfilePair) (
 }
 
 func printAnalyzeJSON(out io.Writer, dockerfiles, images []string) error {
+	if len(dockerfiles) == 0 {
```

Comment:
This condition is used here as well. 
https://github.com/GoogleContainerTools/skaffold/blob/0211c2c7a5deaf2c5f14ae6b4e3581bb72a054d6/pkg/skaffold/initializer/init.go#L124

@dgageot  and @priyawadhwa should we consider failing `skaffold init` if no docker files are present?
AFAIK, `analyze` is used by IDE? What is it used for?

Reply:
Can we get some more context around why `skaffold init` should fail when there are no docker files?

Reply:
`skaffold init` currently fails if there are no Dockerfiles or Kubernetes manifests found, this PR adds the same functionality to `skaffold init --analyze`.

I'm guessing `skaffold init` fails if either is not present because then there is nothing to match either an image or a k8s manifest with when generating the skaffold.yaml. The `--analyze` flag was recently added so that IDEs could support `skaffold init` (ref #1710)

Reply:
It does not right now. Can we move this check to [L112](https://github.com/GoogleContainerTools/skaffold/blob/0211c2c7a5deaf2c5f14ae6b4e3581bb72a054d6/pkg/skaffold/initializer/init.go#L112)
And then remove it from `if c.SkipBuild` flow.

Reply:
Sorry, I thought we only want to see this error `if !c.skipBuild` as written? Because if we're skipping the build section of the skaffold.yaml, it shouldn't matter if there are any Dockerfiles?

In this case, should I check `if !c.SkipBuild`  for `--analyze` as well?

Reply:
yes makes sense. I think we should also change the error message. Technically you don't need a Dockerfile for running skaffold. You can't "create" skaffold config file with sane config without docker files.